RosNet/RosMessageParser/MessageDataParser.cs

```diff
@@ -1,99 +1,95 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using RosNet.Field;
+﻿using RosNet.Field;
 
-namespace RosNet.RosMessageParser
+namespace RosNet.RosMessageParser;
+
+/// <summary>
+/// Parses data in message record
+/// </summary>
+internal static class MessageDataParser
 {
+
     /// <summary>
-    /// Parses data in message record
+    /// Parses messagedata from bytes to fieldvalues using messageDefinition
     /// </summary>
-    public static class MessageDataParser
+    /// <returns>Dictionary contatining the fields in the messageData. Key is name of field and value is the fieldValue</returns>
+    public static Dictionary<string, FieldValue> ParseMessageData(byte[] messageData, List<FieldValue> messageDefinition)
```

Comment:
By making this operate on byte-arrays you end up doing a lot of unnecessary copies of message data here, why can this not operate directly on the `reader` from bagParser? Or at the least take `messageData` as `in`